tests/integration/conftest.py

```diff
@@ -202,6 +202,29 @@ async def postgresql_app_fixture(
         await model.wait_for_idle()
 
 
+@pytest.fixture(scope="module", name="grafana_app_name")
+def grafana_app_name_fixture() -> str:
+    """Return the name of the grafana application deployed for tests."""
+    return "grafana-k8s"
+
+
+@pytest_asyncio.fixture(scope="module", name="grafana_app")
+async def grafana_app_fixture(
+    model: Model,
+    grafana_app_name: str,
+):
+    """Deploy grafana."""
+    app = await model.deploy(
+        "grafana-k8s",
+        application_name=grafana_app_name,
+        channel="latest/edge",
+        trust=True,
+    )
+    await model.wait_for_idle(raise_on_blocked=True)
```

Comment:
Arturo told me to add statuses on each `wait_for_idle` to help with debug so I propose that you do the same thing here.  
This wait could also be specific to grafana since we are in its fixture.

Reply:
Thanks for pointing that out, changed!